### DISCUSSION TITLE
Propose other variant of create_route_map

### DISCUSSION THREAD

**Discussion by SvyatoslavK1** [mobilityhouse/ocpp#463] [Show and tell]
Instead of looping by a fixed list, I propose to use if conditions. I measured with `timeit` and this version works **twice** faster.

```python
    routes = {}
    for attr_name in routables:
        attr = getattr(obj, attr_name)
        if hasattr(attr, "_on_action"):
            action = getattr(attr, "_on_action")
            if action not in routes:
                routes[action] = {}
            routes[action]["_on_action"] = attr
            routes[action]["_skip_schema_validation"] = getattr(
                attr, "_skip_schema_validation", False
            )
        elif hasattr(attr, "_after_action"):
            action = getattr(attr, "_after_action")
            if action not in routes:
                routes[action] = {}
            routes[action]["_after_action"] = attr


    return routes
```

**Comment by OrangeTux** [comment]
Thanks your suggestion!  I like that confirmed the performance increase with `timeit`.

However, I'm hesitant to implement the change. Although I don't have data to back me up, I think `create_route_map()` isn't a hot code path. The function is only called once every time an instance of `ChargePoint` is created. For the use cases I have, the time spend in `create_route_map()` is marginal to the time spend executing the rest of the code.

Therefore, I don't think it's worth the effort and time to change the current implementation.

If you've a different use case than I have and you would get significant benefits by increasing the performance of this piece of code, let me know. I might change my mind.
